### PR TITLE
Add stubs for py_compile.

### PR DESCRIPTION
**Pull request by euresti**

### CODE REVIEW

File: stdlib/2and3/py_compile.pyi [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    exc_type_name = ...  # type: str
+    exc_value = ...  # type: BaseException
+    file = ...  # type: str
+    msg = ...  # type: str
+    def __init__(self, exc_type: str, exc_value: BaseException, file: str, msg: str = ...) -> None: ...
+
+if sys.version_info >= (3, 2):
+    def compile(file: str, cfile: Optional[str] = ..., dfile: Optional[str] = ..., doraise: bool = ..., optimize: int = ...) -> None: ...
+else:
+    def compile(file: str, cfile: Optional[str] = ..., dfile: Optional[str] = ..., doraise: bool = ...) -> None: ...
```
Comment (euresti) [inline stdlib/2and3/py_compile.pyi]: I see some stubs use `unicode`/`Text` for filenames. Should I be doing that here? 

What about in the exceptions?

Comment (ambv) [inline stdlib/2and3/py_compile.pyi]: Don't use "unicode" in 2and3 but if `file` and other arguments in this call can be `bytes` on Python 2, it should be _EitherStr.

**Comment by euresti** [comment]
Fixed the unicode issues.  What I'm not sure of is if I should assume that str can be promoted to Text in python 2, or if I should write 2 versions of `main` since Python3 doesn't support bytes but python 2 supports bytes and unicode.

**Comment by gvanrossum** [comment]
Yes, str can be promoted to Text/unicode in Python 2.

**Pull request merged**